lib/election.js

```diff
@@ -209,35 +209,38 @@ api.isBlockElector = (voter, electors) => {
  * @param callback(err) called once the operation completes.
  */
 api.certify = (ledgerNode, electionTopic, votes, callback) => {
-  // TODO: should be able to see if the vote manifest is already stored
-  //   and skip certification if so
   async.each(votes, (vote, callback) => {
-    _verifyVote(ledgerNode, vote, electionTopic, err => {
-      if(err) {
-        // ignore failed votes, do not store them
-        logger.verbose('Non-critical error in _verifyVote.', err);
-        return callback();
-      }
-      // TODO: reject votes with more than X number of recommended electors
-      // based on continuity config
-      api._hasher(vote, (err, hash) => {
-        if(err) {
-          return callback(err);
+    async.auto({
+      voteHash: callback => api._hasher(vote, callback),
+      voteExists: ['voteHash', (results, callback) =>
+        api._storage.votes.exists(ledgerNode.id, results.voteHash, callback)],
+      verify: ['voteExists', (results, callback) => {
+        if(results.voteExists === true) {
+          return callback();
```

Comment:
Please add a comment above `return callback`:

```js
// vote already exists, do not need to verify or store again
```

Reply:
@dlongley done, ready for another pass.